App/src/components/NewScheduleModal.js

```diff
@@ -0,0 +1,264 @@
+/* eslint-disable no-unused-expressions */
+/** @jsx jsx */
+import { jsx } from "@emotion/core";
+import React, { useState } from "react";
+import Modal from "react-modal";
+import Calendar from "./Calendar";
+import { useUser } from "../contexts/user";
+import { createRequest } from "../services/request";
+import NewCalendar from "../components/NewCalendar";
+import { Button } from "../components/Ui";
+
+const customStyles = {
+  content: {
+    top: "50%",
+    left: "50%",
+    right: "auto",
+    bottom: "auto",
+    transform: "translate(-50%, -50%)"
+  }
+};
+
+function NewScheduleModal({
+  isOpen,
+  onRequestClose,
+  onShiftClick,
+  onPrev,
+  onNext,
+  onToday,
+  startDate,
+  workshiftList,
+  forecast,
+  users
+}) {
+  const user = useUser();
+
+  const [shiftsClicked, setShiftsClicked] = useState([]);
+  const [newMotive, setMotive] = useState("");
+
+  const cancel = {
+    fontSize: "2em",
+    cursor: "pointer",
+    color: "#19216C",
+    fontWeight: "bolder",
+    "&:hover": {
+      color: "#4055A8"
+    }
+  };
+
+  const container = {
+    display: "flex",
+    flexDirection: "column"
+  };
+
+  const head = {
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "space-between"
+  };
+
+  const footer = {
+    display: "flex",
+    justifyContent: "flex-end"
+  };
+
+  const textareaCss = {
+    rows: 4,
+    cols: 50,
+    color: "#35469C",
+    border: "1px solid #35469C"
+    // width: "70%",
+    // marginLeft: "5px"
+  };
+
+  const listCss = {
+    listStyleType: "none",
+    padding: 0,
+    color: "#35469C"
+  };
+
+  const formCss = {
+    display: "flex",
+    flexDirection: "column"
+  };
+
+  function nameFrontDesk(userId) {
+    return users.find(frontdesk => frontdesk.id === parseInt(userId)).name;
+  }
+
+  function clear() {
+    setShiftsClicked([]);
+    setMotive("");
+  }
+
+  function onRequestClear(event) {
+    event.preventDefault();
+    clear();
+  }
+  function onClose(event) {
+    event.preventDefault();
+    clear();
+    onRequestClose();
+  }
+
+  function handleChangeMotive(event) {
+    setMotive(event.target.value);
+  }
+
+  function handleCreateRequest(event) {
+    event.preventDefault();
+    if (!newMotive || shiftsClicked.length < 2) {
+      alert("Must add a motive");
+      return;
+    }
+    const request = {
+      creationDate: new Date(),
+      requester_id: shiftsClicked[0].id,
+      requested_id: shiftsClicked[1].id,
+      date_Shift: shiftsClicked[1].date,
+      current_Shift_id: shiftsClicked[0].shift_id,
+      requested_Shift_id: shiftsClicked[1].shift_id,
+      motive: newMotive
+    };
+
+    try {
+      createRequest(request)
+        .then(() => clear())
+        .then(() => onRequestClose());
+    } catch (error) {
+      console.log(error.message);
```

Comment:
```suggestion
```

Instead of only logging the error you should do something, in which cases a request could fail? you may want to notify the user there was an error and what was the error

Reply:
Thanks for your advice, We will implement in the next pullrequest.